extensions/undertow/runtime/src/main/java/io/quarkus/undertow/runtime/KnownPathResourceManager.java

```diff
@@ -118,18 +125,38 @@ public boolean isDirectory() {
         @Override
         public List<Resource> list() {
             List<Resource> ret = new ArrayList<>();
-            String slashPath = path + "/";
-            for (String i : files.headSet(path)) {
-                if (i.startsWith(slashPath)) {
-                    try {
-                        ret.add(underlying.getResource(i));
-                    } catch (IOException e) {
-                        throw new RuntimeException(e);
+            String slashPath = path.isEmpty() ? path : path + "/";
+            if (IS_WINDOWS) {
+                slashPath = slashPath.replaceAll("/", "\\\\"); // correct Windows paths
+            }
+            SortedSet<String> fileSet = files.tailSet(slashPath);
+            SortedSet<String> dirSet = directories.tailSet(slashPath);
+
+            for (var s : List.of(fileSet, dirSet)) {
+                for (String i : s) {
+                    if (i.equals(slashPath)) {
+                        continue;
+                    }
+                    if (i.startsWith(slashPath)) {
+                        i = evaluatePath(i);
+                        if (!i.substring(slashPath.length()).contains("/")) {
+                            try {
+                                Resource resource = underlying.getResource(i);
+                                if (resource == null) {
+                                    throw new RuntimeException("Unable to get listed resource " + i + " from directory " + path
+                                            + " for path " + slashPath + " from underlying manager " + underlying);
+                                }
+                                ret.add(resource);
+                            } catch (IOException e) {
+                                throw new UncheckedIOException(e);
```

Comment:
I think you're missing an import here?

Reply:
Weird its not in the original code or in the PR I started with from @stuartwdouglas but seems to work?